src/page-response-impact/hooks.tsx

```diff
@@ -0,0 +1,186 @@
+import { useEffect, useState } from "react";
+
+import { getFacilities } from "../database";
+import {
+  EpidemicModelState,
+  getLocaleDefaults,
+} from "../impact-dashboard/EpidemicModelContext";
+import { CurveFunctionInputs } from "../infection-model";
+import { LocaleData } from "../locale-data-context";
+import { Facilities, Scenario } from "../page-multi-facility/types";
+import {
+  calculateCurveData,
+  getCurveInputs,
+  getModelInputs,
+  getSystemWideSums,
+  originalProjection,
+} from "./responseChartData";
+import {
+  buildReductionData,
+  buildResponseImpactCardData,
+  reductionCardDataType,
+} from "./utils/ResponseImpactCardStateUtils";
+
+type FacilitiesState = {
+  data: Facilities;
+  loading: boolean;
+};
+
+type SystemWideData = {
+  hospitalBeds: number;
+  staffPopulation: number;
+  prisonPopulation: number;
+};
+
+export function useFacilities(
+  scenarioId: string,
+  localeDataSource: LocaleData,
+) {
+  const [facilities, setFacilities] = useState<FacilitiesState>({
+    data: [],
+    loading: true,
+  });
+
+  useEffect(() => {
+    async function fetchFacilities() {
+      if (!scenarioId) return;
+      const facilitiesData = await getFacilities(scenarioId);
+      if (facilitiesData) {
+        setFacilities({
+          data: facilitiesData,
+          loading: false,
+        });
+      }
+    }
+
+    fetchFacilities();
+  }, [scenarioId, localeDataSource]);
+
+  return facilities;
+}
+
+export function useModelInputs(
+  facilities: FacilitiesState,
+  localeDataSource: LocaleData,
+) {
+  const [modelInputs, setModelInputs] = useState([] as EpidemicModelState[]);
+
+  useEffect(() => {
+    if (facilities.data.length) {
+      const modelInputs = getModelInputs(facilities.data, localeDataSource);
+      setModelInputs(modelInputs);
+    }
+  }, [facilities, localeDataSource]);
+
+  return modelInputs;
+}
+
+export function useCurrentCurveData(
+  modelInputs: EpidemicModelState[],
+  localeDataSource: LocaleData,
+) {
+  const [currentCurveInputs, setCurrentCurveInputs] = useState(
+    [] as CurveFunctionInputs[],
+  );
+
+  useEffect(() => {
+    if (modelInputs.length === 0) return;
+
+    const currentCurveInputs = getCurveInputs(modelInputs);
+    setCurrentCurveInputs(currentCurveInputs);
+  }, [modelInputs, localeDataSource]);
+
+  return currentCurveInputs;
+}
+
+export function useSystemWideData(
+  scenarioPopulations: Scenario["populations"],
+  modelInputs: EpidemicModelState[],
+  localeDataSource: LocaleData,
+) {
+  const [systemWideData, setSystemWideData] = useState<SystemWideData>({
+    hospitalBeds: 0,
+    staffPopulation: 0,
+    prisonPopulation: 0,
+  });
+
+  useEffect(() => {
+    if (modelInputs.length === 0) return;
+
+    // Reverse the populations array to get the most recently modelled populations
+    const {
+      staffPopulation: userInputStaffPopulation,
+      incarceratedPopulation: userInputIncarceratedPopulation,
+    } = reverse(scenarioPopulations)[0];
```

Comment:
I'm getting an error here since I do not have any `scenarioPopulations` (`scenario.populations`). Do we need to have a check for its `length` or give it a default value? 

Or do we have a migration strategy for this? My facility data is pre scenario-UI.

Reply:
when I provide some data for `scenarioPopulations`, I get an infinite looping of the the following 3 hooks:

- useSystemWideData
- useOriginalCurveData
- useReductionData

I believe it is due to lodash's `reverse` mutating the array in place https://lodash.com/docs/4.17.15#reverse while `scenarioPopulations` is also in the dependency array for this `useEffect` hook.

Reply:
For 1. that's a good point, I will add a guard for that. 
For 2. good catch--I hadn't noticed that locally, I will fix that

Thanks for reviewing!